hecuba_py/hecuba/hnumpy.py

```diff
@@ -176,6 +276,8 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
             kwargs['out'] = tuple(out_args)
         else:
             outputs = (None,) * ufunc.nout
+        if self._is_persistent and len(self.shape):
+            self._hcache.load_numpy_slices([self.storage_id], [self.view(np.ndarray)], None)
```

Comment:
This loads all the data without considering that maybe it was already in memory?

Reply:
What about if we check it with the following condition:
if self._is_persistent and len(self.shape) and np.all(self.view(np.ndarray) == 0):
           self._hcache.load_numpy_slices([self.storage_id], [self.view(np.ndarray)], None)
           self._hcache.store_numpy_slices([self.storage_id], [self.view(np.ndarray)], None)

Reply:
The user may be working with an array of zeros, in this case it wouldn't work. Why don't you use the attribute `loaded_coordinates`? 

I am concerned about reading the data, we can leave the store_numpy_slices as it is for now :)

Reply:
i found that the attribute _numpy_full_loaded can be helpful:

        if self._is_persistent and len(self.shape) and self._numpy_full_loaded == False:
            self._hcache.load_numpy_slices([self.storage_id], [self.view(np.ndarray)], None)